src/App.tsx

```diff
@@ -15,48 +17,126 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortingParams {
+  Name = 'name',
+  Length = 'length',
+  Default = '',
+}
+
+interface SortOptions {
+  option: SortingParams,
+  reversed: boolean,
+}
+
+const getVisibleGoods = (
+  goods: typeof goodsFromServer,
+  { option, reversed }: SortOptions,
+) => {
+  let visibleGoods = goods;
+
+  if (option) {
+    visibleGoods = [...visibleGoods].sort((good1, good2) => {
```

Comment:
don't need to overwrite the visibleGoods, because method sort will do that
```suggestion
  let visibleGoods = [...goods];

  if (option) {
    visibleGoods.sort((good1, good2) => {
```

Reply:
But it will broke down my conditional rendering in 116 line, what should I do in this case?